From 360e1b135034e3c2dadd65b001f24f883ae55019 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:36 -0400
Subject: [ethernet] igb: Start temperature sensor attribute index with 1

Message-id: <1409062359-13063-19-git-send-email-sassmann@redhat.com>
Patchwork-id: 89132
O-Subject: [RHEL7.1 PATCH 18/81] igb: Start temperature sensor attribute index with 1
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Guenter Roeck <linux@roeck-us.net>

Per hwmon ABI, temperature sensor attribute index starts with 1, not 0.

Signed-off-by: Guenter Roeck <linux@roeck-us.net>
Reviewed-by: Jean Delvare <khali@linux-fr.org>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e6e25bba9b8a343af6b5e281a0378d3c8ae52165)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_hwmon.c b/drivers/net/ethernet/intel/igb/igb_hwmon.c
index 58f1ce9..d15bdfa 100644
--- a/drivers/net/ethernet/intel/igb/igb_hwmon.c
+++ b/drivers/net/ethernet/intel/igb/igb_hwmon.c
@@ -124,22 +124,22 @@ static int igb_add_hwmon_attr(struct igb_adapter *adapter,
  case IGB_HWMON_TYPE_LOC:
   igb_attr->dev_attr.show = igb_hwmon_show_location;
   snprintf(igb_attr->name, sizeof(igb_attr->name),
-    "temp%u_label", offset);
+    "temp%u_label", offset + 1);
   break;
  case IGB_HWMON_TYPE_TEMP:
   igb_attr->dev_attr.show = igb_hwmon_show_temp;
   snprintf(igb_attr->name, sizeof(igb_attr->name),
-    "temp%u_input", offset);
+    "temp%u_input", offset + 1);
   break;
  case IGB_HWMON_TYPE_CAUTION:
   igb_attr->dev_attr.show = igb_hwmon_show_cautionthresh;
   snprintf(igb_attr->name, sizeof(igb_attr->name),
-    "temp%u_max", offset);
+    "temp%u_max", offset + 1);
   break;
  case IGB_HWMON_TYPE_MAX:
   igb_attr->dev_attr.show = igb_hwmon_show_maxopthresh;
   snprintf(igb_attr->name, sizeof(igb_attr->name),
-    "temp%u_crit", offset);
+    "temp%u_crit", offset + 1);
   break;
  default:
   rc = -EPERM;
-- 
1.7.1